test/NuGet.Clients.Tests/NuGet.PackageManagement.UI.Test/NuGetPackageDetailsTest.cs

```diff
@@ -0,0 +1,86 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using NuGet.Versioning;
+using Xunit;
+
+namespace NuGet.PackageManagement.UI.Test
+{
+    public class NuGetPackageDetailsTest
+    {
+        [Theory]
+        [InlineData("Newtonsoft.Json", "1.2.3")]
+        [InlineData("Moq", "2.3.4")]
+        public void NuGetPackageDetails_BothPropertiesNotNull_NotNull(string name, string version)
+        {
+            //Arrange
+            NuGetVersion nugetVersion;
+            NuGetVersion.TryParse(version, out nugetVersion);
+
+            //Act
+            var packagedetails = new NuGetPackageDetails(name, nugetVersion);
+
+            //Assert
+            string actualPackageName = packagedetails.PackageName;
+            NuGetVersion actualVersion = packagedetails.VersionNumber;
+            Assert.Equal(name, actualPackageName);
+            Assert.Equal(nugetVersion, actualVersion);
+        }
+
+        [Theory]
+        [InlineData(null, "1.2.3")]
+        [InlineData(null, "2.3.4")]
+        public void NuGetPackageDetails_NameNullVersionNotNull_NotNull(string name, string version)
+        {
+            //Arrange
+            NuGetVersion nugetVersion;
+            NuGetVersion.TryParse(version, out nugetVersion);
+
+            //Act
+            var packagedetails = new NuGetPackageDetails(name, nugetVersion);
+
+            //Assert
+            string actualPackageName = packagedetails.PackageName;
+            NuGetVersion actualVersion = packagedetails.VersionNumber;
+            Assert.Null(actualPackageName);
+            Assert.Equal(nugetVersion, actualVersion);
+        }
+
+        [Theory]
+        [InlineData("Newtonsoft.Json", null)]
+        [InlineData("Moq", null)]
+        public void NuGetPackageDetails_NameNotNullVersionNull_NotNull(string name, string version)
```

Comment:
The "_NotNull" part at the end of this test method name doesn't apply I think. I think that was when you checking if the instantiated object was not null after it was instantiated.